spec/unit/ui_spec.rb

```diff
@@ -57,7 +65,7 @@
       allow($stdout).to receive(:tty?).and_return(true)
       allow(LocalSystem).to receive(:validate_existence_of_package).
         and_raise(Machinery::Errors::MissingRequirement)
-      expect($stdout).to receive(:puts).with(output)
+      expect($stdout).to receive(:print).with(output + "\n")
```

Comment:
This test is not ideal because it tests the implementation, not the result. You can tell that because you needed to change the test, although the behavior of the method hasn't changed. It would be better to write the test in a way that is more independent of the implementation.

I would be ok with leaving that to a later refactoring, though.